src/gui/maptools/qgsmaptoolcapture.cpp

```diff
@@ -316,8 +316,8 @@ bool QgsMapToolCapture::tracingAddVertex( const QgsPointXY &point )
   mSnappingMatches.removeLast();
   mSnappingMatches.append( QgsPointLocator::Match() );
 
-  int pointBefore = mCaptureCurve.numPoints();
   addCurve( new QgsLineString( mapPoints ) );
+  int pointBefore = mCaptureCurve.numPoints();
```

Comment:
Just to make it clearer what this is actually doing, could we move all the logic relating to pointBefore/pointAfter and the sync snapping matches into the block which starts with `if ( QgsSettingsRegistryCore::settingsDigitizingConvertToCurve->value() )` ? As far as I can see this logic ONLY applies now if the curve approximation is happening.

And on that note... if the unsegmentized line ends up with LESS vertices then the segmentized one, won't `mSnappingMatches` have too many entries and need to be trimmed? Maybe this whole logic could be simplified down to just:

```
    // sync the snapping matches list
    mSnappingMatches.resize( mCaptureCurve.numPoints() )
```

Reply:
Good catch! Done c3c3ccc8812